crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -94,7 +94,36 @@ pub(crate) fn type_check_expression(
         HirExpression::Index(index_expr) => {
             type_check_index_expression(interner, index_expr, errors)
         }
-        HirExpression::Call(call_expr) => {
+        HirExpression::Call(mut call_expr) => {
+            if let Some(meta) = interner.try_function_meta(&call_expr.func_id) {
+                if meta.kind == crate::FunctionKind::LowLevel {
+                    let attribute = meta.attributes.expect("all low level functions must contain an attribute which contains the opcode which it links to");
+                    let opcode = attribute.foreign().expect(
+                        "ice: function marked as foreign, but attribute kind does not match this",
+                    );
+                    if !interner.foreign(&opcode) {
+                        if let Some(func_id2) = interner.get_alt(opcode) {
+                            let expr2 = expr::HirCallExpression {
+                                func_id: func_id2,
+                                arguments: call_expr.arguments.clone(),
+                                alt: None,
```

Comment:
I think we can mutate the current HirCallExpression to avoid cloning the arguments here

Reply:
Done